### DISCUSSION TITLE
Control the scope of action of the var_to_str

### DISCUSSION THREAD

**Discussion by CodeLazier** [godotengine/godot-proposals#7457] [Scripting]
gdscript has no private member field, so serialization functions like var_to_str will write all private fields of a class to string, which is inelegant and whether or not to accept a parameter such as a field starting with "_" is automatically ignored serialization

**Comment by dalexeev** [comment]
It looks like a bug, it is inconsistent with binary serialization and `*.tscn`/`*.tres`.

https://github.com/godotengine/godot/blob/bceac8c34f2cd6f9f156b3b1b9cbd012bc45d928/core/variant/variant_parser.cpp#L1917-L1922

https://github.com/godotengine/godot/blob/bceac8c34f2cd6f9f156b3b1b9cbd012bc45d928/core/io/marshalls.cpp#L1495-L1502

As a workaround, in 4.2 dev you can use `_validate_property()` to change usage property flags.

See also godotengine/godot#80585.
